### DISCUSSION TITLE
How to write slugify middleware

### DISCUSSION THREAD

**Discussion by meotimdihia** [prisma/prisma#6479] [Q&A]
I want to write middleware update `slug` field:

```javascript
prisma.$use(async (params, next) => {
//.........
      if (
        ["create", "update", "upsert", "updateMany"].indexOf(params.action) != -1 &&
        params.model == "md_people"
      ) {
        const slugify = require("@sindresorhus/slugify")
        
        if (params.args.data != undefined) {
          params.args.data['slug'] = slugify(params.args.data['title'])
        }
      }

//.........
    })
```

I got **the problem**: *the `slug` field is required.* 
![Code_2021-04-09_18-19-44](https://user-images.githubusercontent.com/300961/114172692-2e710100-9960-11eb-8451-ad03c0c8e582.png)

Do we have a legit method for this problem, cause the slug field will be updated in middle?

**Comment by ryands17** [comment]
@meotimdihia 👋 
In this case, you would need to pass an empty string to `slug` as it is required. The middleware is adding it but the value must also be defined where you're calling the query.

Either that or make `slug` optional.

**Reply by meotimdihia** [reply]
Thanks for the answer.  I think about this solution but I want to know if we have a better way :)

**Reply by ryands17** [reply]
You can add `// @ts-ignore` to the code where you are passing in the data and then add it in the middleware so it will work then.

**Reply by ryands17** [reply]
With a model like this:

```prisma
model User {
  id   Int    @id @default(autoincrement())
  name String
}
```

Here `name` is required, but I can pass it via middleware as follows:

```ts
prisma.$use(async (params, next) => {
  if (params.model === 'User') {
    params.args.data.name = 'user 1'
    return next(params)
  }
})

async function main() {
  // @ts-ignore
  await prisma.user.create({ data: {} })
}
```

**Reply by meotimdihia** [reply]
> You can add `// @ts-ignore` to the code where you are passing in the data and then add it in the middleware so it will work then.

I prefer this solution. The empty string makes the code looks confuse :).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
